### ISSUE TITLE
Draw text that doesn't overlap

### ISSUE THREAD

**Issue by heinermann** [bwapi/bwapi#547]
The ability to draw text to the screen that doesn't overlap. This would probably be as a new function that duplicates the drawText API.

It would be useful for the Tournament module in the stream, as well as so bot developers don't have to play with text positions.

Any naming ideas? drawTextClosest, drawTextClean, drawTextAway, drawTextNoOverlap...

Or should I change the behaviour of drawText, and create a new function drawTextOverlap? 

![screenshot from 2015-02-02 19 15 17](https://cloud.githubusercontent.com/assets/1966366/6011978/f7c4e2e2-ab0f-11e4-9beb-8d3665d02e8b.png)

**Comment by albertouri** [comment]
For naming ideas maybe drawTextOnTop.
But personally I would implement different "draw text layers". This way a Tournament module can print in one layer and disable all the other layers.

**Comment by heinermann** [comment]
I was thinking that the text would be automatically relocated to not overlap. Maybe provide a priority value to indicate which pieces of text have bidding preference over others, where the maximum bid amount for AI modules is say 1000, and maximum for tournament modules is max_int.

**Comment by albertouri** [comment]
And what happen if you don't have space to relocate the text?
Can other draw elements like lines or dots overlap text?

**Comment by heinermann** [comment]
Yes it's possible that other elements can overlap text. We could also give them priority values. When there's not enough space it can just overlap anyway.

Maybe drawTextRelocate is a better name. Also perhaps we can have a transparency colour that erases stuff an AI module has drawn (for tournament module).

There are different solutions here I'm just not sure what would work best.
